### PR TITLE
get irq->module relationship from /sys/bus/pci/*/driver

### PR DESCRIPTION
**Pull request by liuchao173**
https://github.com/Irqbalance/irqbalance/issues/202

Signed-off-by: Chao Liu <liuchao173@huawei.com>

### CODE REVIEW

File: classify.c [hunk 1]
```diff
@@ -604,19 +607,42 @@ static int check_for_irq_ban(int irq, GList *proc_interrupts)
...
 
+static void get_mod_name(char *path, char *mod)
+{
+	char *cmd;
+	FILE *output;
+
+	if (!path)
+		return;
+
+	cmd = alloca(strlen(path) + 30);
```
Comment (nhorman) [inline classify.c:618]: why +30 here?

File: classify.c [hunk 2]
```diff
@@ -594,6 +594,9 @@ static int check_for_irq_ban(int irq, GList *proc_interrupts)
 	/*
 	 * Check to see if we banned module which the irq belongs to.
 	 */
+	if (strlen(mod) > 0 && check_for_module_ban(mod))
```
[the remark this answers is not in the document]
Comment (liuchao173) [reply classify.c:597]: mod is defined in add_new_irq as char array，it won't be null

File: classify.c [hunk 3]
```diff
@@ -604,19 +607,42 @@ static int check_for_irq_ban(int irq, GList *proc_interrupts)
...
+	if (!cmd)
+		return;
+
+	sprintf(cmd, "cat %s/uevent | grep \"DRIVER=\"", path);
+	output = popen(cmd, "r");
+	if (!output)
+		return;
+
+	fgets(mod, 128, output);
+	pclose(output);
```
Comment (nhorman) [inline classify.c:628]: why go to all the trouble of fishing the DRIVER field out of the event file?  you can find the same information via basename(readlink(/sys/bus/pci/*/driver)

File: classify.c (hunk 1 above)
Comment (liuchao173) [reply classify.c:618]: It is prepared for the remaining string "cat %s/uevent | grep \"DRIVER=\""

File: classify.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (nhorman) [reply classify.c:597]: thats even worse, by defining it on the stack without memsetting it to anything, you're passing around whatever garbage is on the stack when the call is made, so you avoid the semi-guaranteed segfault, in exchange for a random segfault that occurs when strlen indexes through whatever random memory you have that likely doesn't have a NULL terminator.

File: classify.c (hunk 1 above)
Comment (nhorman) [reply classify.c:618]: yeah, but 30?  Thats just a random sizing of the string.  at least define your template and take its strlen, i.e.:
#define CMDTMPL "/uevent | grep \"DRIVER=\""
...
cmd = alloca(strlen(path) + strlen(CMDTMPL) + 1); //alloc cmd template for path + null terminator byte

File: classify.c (hunk 3 above)
Comment (liuchao173) [reply classify.c]: done

**Pull request merged**